plugins/nexus-repository-manager/src/components/ArtifactTable/ArtifactTable.tsx

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+
+import { Link, Table, type TableColumn } from '@backstage/core-components';
+
+import { Box, Chip, makeStyles } from '@material-ui/core';
+
+import { formatByteSize } from '@janus-idp/shared-react';
+
+import type { AssetHash } from '../../types';
+
+export type ArtifactRowData = {
+  id?: string;
+  version?: string;
+  artifact?: string;
+  assetVariants: Set<string>;
+  repositoryType?: string;
+  hash?: AssetHash;
+  lastModified?: string;
+  sizeBytes?: number;
+};
+
+const useStyles = makeStyles(theme => ({
+  chip: {
+    margin: 0,
+    marginRight: '.2em',
+    height: '1.5em',
+    '& > span': {
+      padding: '.3em',
+    },
+  },
+  empty: {
+    padding: theme.spacing(2),
+    display: 'flex',
+    justifyContent: 'center',
+  },
+}));
+
+export const ArtifactTable = ({
+  artifacts,
+  title,
+}: {
+  artifacts: ArtifactRowData[];
+  title: string;
+}) => {
+  const classes = useStyles();
+
+  const columns: TableColumn<ArtifactRowData>[] = [
+    {
+      title: 'Version',
+      field: 'version',
+      type: 'string',
+      highlight: true,
+    },
+    {
+      title: 'Artifact',
+      field: 'artifact',
+      type: 'string',
+      render: rowData => (
+        <>
+          {rowData.artifact}
+          <Box
+            sx={{
+              display: 'flex',
+              alignItems: 'center',
+              flexWrap: 'wrap',
+              marginTop: '0.2em',
+            }}
+          >
+            {/* sort/reverse for stable order, and so we get `jar +sources` */}
```

Comment:
this seems like a bit of a hack, but I think it's fine? having all `+classifiers` at the end, and then the rest with *a* stable sort order seems good enough

Reply:
This should be fine; however, if it takes too long to load we may need to change it to `.reduce()` instead of `.sort().reverse().map()` because of the time complexity. I am assuming it won't be an issue here because `assetVariants` should be reasonably sized.